contracts/5/IbETHRouter.sol

```diff
@@ -62,27 +58,49 @@ library TransferHelper {
     }
 }
 
-contract UniswapV2Router02 is IUniswapV2Router02 {
+// IbETHRouter modifies UniswapV2Router02 to use AlphaHomora's ibETH, instead of WETH
+contract IbETHRouter {
     using SafeMath for uint256;
 
     address public factory;
-    address public WETH;
+    address payable ibETH; // usd ibETH, instead of WETH
 
     modifier ensure(uint256 deadline) {
-        require(deadline >= block.timestamp, "UniswapV2Router: EXPIRED");
+        require(deadline >= block.timestamp, "IbETHRouter: EXPIRED");
         _;
     }
-
-    constructor(address _factory, address _WETH) public {
+    
+    constructor(address _factory, address payable _ibETH) public {
         factory = _factory;
-        WETH = _WETH;
+        ibETH = _ibETH;
     }
 
     function() external payable {
-        assert(msg.sender == WETH); // only accept ETH via fallback from the WETH contract
-    }
-
-    // **** ADD LIQUIDITY ****
+        assert(msg.sender == ibETH); // only accept ETH via fallback from the Bank contract
+    }
+
+    // **** ETH-ibETH FUNCTIONS ****
+    // Get number of ibETH received for the amouthETH send to Bank
+    function ETHToIbETH(uint256 amountETH) public view returns (uint256) {
```

Comment:
Let's name this `exactETHToIbETH` (exactly this ETH and convert to how much ib eth)